causalnex/structure/data_generators.py

```diff
@@ -626,3 +611,336 @@ def _raise_dist_error(name: str, dist: str, dist_options):
+        w_decay: exponent of weights decay for slices that are farther apart. Default is 1.0, which implies no decay
+
+    Returns:
+        G_inter: weighted, bipartite DAG for inter-slice connections
+
+    Raises:
+        ValueError: if graph type not known
+    """
+    if w_min > w_max:
+        raise ValueError(
+            "Absolute minimum weight must be less than or equal to maximum weight: {} > {}".format(
+                w_min, w_max
+            )
+        )
+
+    if graph_type == "erdos-renyi":
+        prob = float(degree) / num_nodes
+        b = (np.random.rand(p * num_nodes, num_nodes) < prob).astype(float)
+        # np.fill_diagonal(B, 0)  # get rid of autoregressive terms
+    elif graph_type == "full":  # ignore degree, only for experimental use
+        b = np.ones([p * num_nodes, num_nodes])
+    else:
+        raise ValueError("Unknown inter-slice graph type")
+    u = []
+    for i in range(p):
+        u_i = np.random.uniform(low=w_min, high=w_max, size=[num_nodes, num_nodes]) / (
+            w_decay ** i
+        )
+        u_i[np.random.rand(num_nodes, num_nodes) < 0.5] *= -1
+        u.append(u_i)
+    if u:
+        u = np.concatenate(u, axis=0)
+    else:
+        u = np.array([]).reshape(b.shape)
+
+    a = (b != 0).astype(float) * u
+
+    df = pd.DataFrame(
+        a,
+        index=[
+            "{var}_lag{l_val}".format(var=var, l_val=l_val)
+            for l_val in range(1, p + 1)
+            for var in range(num_nodes)
+        ],
+        columns=[
+            "{var}_lag{l_val}".format(var=var, l_val=0) for var in range(num_nodes)
+        ],
+    )
+    idxs = list(df.index)
+    cols = list(df.columns)
+    for i in idxs:
+        df[i] = 0
+    for i in cols:
+        df.loc[i, :] = 0
+
+    g_inter = StructureModel(df)
+    return g_inter
+
+
+def generate_dataframe_dynamic(  # pylint: disable=R0914
+    g: StructureModel,
+    n_samples: int = 1000,
+    burn_in: int = 100,
+    sem_type: str = "linear-gauss",
+    noise_scale: float = 1.0,
+    drift: np.ndarray = None,
+) -> pd.DataFrame:
+    """Simulate samples from dynamic SEM with specified type of noise.
+    Args:
+        g: Dynamic DAG
+        n_samples: number of samples
+        burn_in: number of samples to discard
+        sem_type: {linear-gauss,linear-exp,linear-gumbel}
+        noise_scale: scale parameter of noise distribution in linear SEM
+        drift: array of drift terms for each node, if None then the drift is 0
+    Returns:
+        X: [n,d] sample matrix, row t is X_t
+        Y: [n,d*p] sample matrix, row t is [X_{t-1}, ..., X_{t-p}]
+    Raises:
+        ValueError: if sem_type isn't linear-gauss/linear_exp/linear-gumbel
+    """
+    if sem_type not in ("linear-gauss", "linear-exp", "linear-gumbel"):
+        raise ValueError("unknown sem type")
+    intra_nodes = sorted([el for el in g.nodes if "_lag0" in el])
+    inter_nodes = sorted([el for el in g.nodes if "_lag0" not in el])
```

Comment:
```suggestion
    intra_nodes = sorted(el for el in g.nodes if "_lag0" in el)
    inter_nodes = sorted(el for el in g.nodes if "_lag0" not in el)
```

`sorted` takes an iterable, so no need to build a list